ArduPlane/ArduPlane.cpp

```diff
@@ -968,6 +969,81 @@ void Plane::update_flight_stage(void)
     airspeed.set_EAS2TAS(barometer.get_EAS2TAS());
 }
 
+/*
+*  ArduSoar support function
+*
+*  Peter Braswell and Samuel Tabor
+*/
+void Plane::update_soaring() {
+    
+    if (!soaring_controller.is_active()) {
+        return;
```

Comment:
The way you have it now where it just stops serving the system doesn't clear any situations where if its on a loiter and then you disable the feature and resume auto mission.. if you reenable it a few minutes would you continue loitering? How about other scenarios?